cmd/sovereignnode/chainSimulator/tests/bridge/incoming_test.go

```diff
@@ -0,0 +1,138 @@
+			BypassTxSignatureCheck: false,
+			TempDir:                t.TempDir(),
+			PathToInitialConfig:    defaultPathToInitialConfig,
+			NumOfShards:            1,
+			GenesisTimestamp:       time.Now().Unix(),
+			RoundDurationInMillis:  uint64(6000),
+			RoundsPerEpoch:         core.OptionalUint64{},
+			ApiInterface:           api.NewNoApiInterface(),
+			MinNodesPerShard:       2,
+			ConsensusGroupSize:     2,
+		},
+	})
+	require.Nil(t, err)
+	require.NotNil(t, cs)
+
+	defer cs.Close()
+
+	nodeHandler := cs.GetNodeHandler(core.SovereignChainShardId)
+
+	logger.SetLogLevel("*:DEBUG,process:TRACE,block:TRACE")
+
+	headerNonce := uint64(9999999)
+	randomHeader := createHeaderV2(headerNonce, generateRandomHash(), generateRandomHash())
+
+	err = cs.GenerateBlocks(1)
+	require.Nil(t, err)
+
+	incomingHeader := createIncomingHeader(nodeHandler, &headerNonce, randomHeader, nil)
+	err = nodeHandler.GetIncomingHeaderSubscriber().AddHeader(generateRandomHash(), incomingHeader)
+	require.Nil(t, err)
+
+	err = cs.GenerateBlocks(1)
+	require.Nil(t, err)
+
+	incomingHeader = createIncomingHeader(nodeHandler, &headerNonce, incomingHeader.Header, createTransactionsEvent())
+	err = nodeHandler.GetIncomingHeaderSubscriber().AddHeader(generateRandomHash(), incomingHeader)
+	require.Nil(t, err)
+
+	err = cs.GenerateBlocks(1)
+	require.Nil(t, err)
+
+	incomingHeader = createIncomingHeader(nodeHandler, &headerNonce, incomingHeader.Header, nil)
+	err = nodeHandler.GetIncomingHeaderSubscriber().AddHeader(generateRandomHash(), incomingHeader)
+	require.Nil(t, err)
+
+	err = cs.GenerateBlocks(1)
+	require.Nil(t, err)
+
+	address := "erd1crq888sv7c3j4y6d9etvlngs3q0tr3endufgls245j5y9yk0ulmqlsuute"
+	account, _, err := nodeHandler.GetFacadeHandler().GetAccount(address, coreAPI.AccountQueryOptions{})
+	require.Nil(t, err)
+	require.NotNil(t, account)
+
+	esdts, _, err := nodeHandler.GetFacadeHandler().GetAllESDTTokens(address, coreAPI.AccountQueryOptions{})
+	require.Nil(t, err)
+	require.NotNil(t, esdts)
+	require.True(t, len(esdts) > 0)
+}
+
+func createIncomingHeader(nodeHandler process.NodeHandler, headerNonce *uint64, prevHeader *block.HeaderV2, txsEvent []*transaction.Event) *sovereign.IncomingHeader {
+	*headerNonce++
+	prevHeaderHash, _ := core.CalculateHash(nodeHandler.GetCoreComponents().InternalMarshalizer(), nodeHandler.GetCoreComponents().Hasher(), prevHeader)
+	incomingHeader := &sovereign.IncomingHeader{
+		Header:         createHeaderV2(*headerNonce, prevHeaderHash, prevHeader.GetRandSeed()),
+		IncomingEvents: txsEvent,
+	}
+	return incomingHeader
+}
+
+func createHeaderV2(nonce uint64, prevHash []byte, prevRandSeed []byte) *block.HeaderV2 {
+	return &block.HeaderV2{
+		Header: &block.Header{
+			PrevHash:     prevHash,
+			Nonce:        nonce,
+			Round:        nonce,
+			RandSeed:     generateRandomHash(),
+			PrevRandSeed: prevRandSeed,
+			ChainID:      []byte(configs.ChainID),
+		},
+	}
+}
+
+func createTransactionsEvent() []*transaction.Event {
```

Comment:
This is too hard-coded and not usable in the future. You already have access here to dataCodec in sovereign package.
Let's make a more generic function which is capable of generating those "magic" hex characters based on smth like:
- rcv addr
- token to deposit
- amount to deposit
- etc.